src/couchConfig.js

```diff
@@ -0,0 +1,357 @@
+/*
+Copyright 2017 OCAD University
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://raw.githubusercontent.com/fluid-project/fluid-couch-config/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion");
+var isEqual = require("underscore").isEqual;
+var size = require("underscore").size;
+
+fluid.defaults("fluid.couchConfig.base", {
+    gradeNames: ["fluid.component"],
+    dbConfig: {
+        couchURL: "http://localhost:5984"
+    }
+});
+
+fluid.defaults("fluid.couchConfig.db", {
+    gradeNames: ["fluid.couchConfig.base"],
+    dbConfig: {
+        // dbName: "targetDB",
+    },
+    events: {
+        // Fired after the component confirms the target DB exists;
+        // necessary for sequencing document-related updates
+        onDBExists: null
+    },
+    invokers: {
+        ensureDBExists: {
+            funcName: "fluid.couchConfig.db.ensureDBExists",
+            args: ["{that}.options.dbConfig.couchURL", "{that}.options.dbConfig.dbName", "{that}.events.onDBExists"]
+        }
+    }
+});
+
+fluid.couchConfig.db.ensureDBExists = function (couchURL, dbName, completionEvent) {
+    console.log("Making sure DB " + dbName + " exists in Couch instance at " + couchURL);
+    var nano = require("nano")(couchURL);
+    nano.db.get(dbName, function (err, body) {
+        if (!err) {
+            console.log("DB " + dbName + " exists");
+            completionEvent.fire();
+        } else {
+            if (err.statusCode === 404) {
+                console.log("DB does not exist, trying to create");
+                nano.db.create(dbName, function (err, body) {
+                    if (!err) {
+                        console.log("DB " + dbName + " created");
+                        completionEvent.fire();
+                    } else {
+                        console.log("DB " + dbName + " could not be created");
```

Comment:
Once we have a mechanism in place for reporting errors programmatically (through promises, events), we should be able to reduce the amount of `console.log()` calls. And any remaining logging that we want to keep in place for debugging, we should switch to `fluid.log()`.

Reply:
Proper error logging via promises is now available, so the number of calls to `console.log()` has been reduced. When the code is switched to `fluid.log()`, though, no feedback is given when running the `couchConfigBasic` example setup. I.e. the various step-by-step updates are hidden, and only messages passed back to the `onError` listener are displayed.

Reply:
By default, `fluid.log()` logs at the `INFO` logLevel priority, as described in the table at https://docs.fluidproject.org/infusion/development/CoreAPI.html#fluidloglevel.

This means plain `fluid.log()` messages won't be displayed unless a more verbose log level than the default is set - see the description of the logging behaviour at https://docs.fluidproject.org/infusion/development/CoreAPI.html#fluidsetloggingloglevel

Try the following and see if it results in the step-by-step update: 
* `fluid.log(message, fluid.logLevel.IMPORTANT)` when logging the message
* `fluid.setLogging(true)`

Reply:
Thanks for your help, @waharnum, using `fluid.setLogging(true)` has solved the issue, and the logging is much cleaner looking with the timestamps, now. I've made a commit which incorporates this change.